chains/orchestrator-paras/runtime/common/src/lib.rs

```diff
@@ -14,6 +14,37 @@
 // You should have received a copy of the GNU General Public License
 // along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
 #![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::{fungible::Credit, tokens::imbalance::ResolveTo, OnUnbalanced};
+use pallet_balances::NegativeImbalance;
 #[cfg(feature = "runtime-benchmarks")]
 pub mod benchmarking;
 pub mod migrations;
+
+pub struct DealWithFees<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<Credit<R::AccountId, pallet_balances::Pallet<R>>> for DealWithFees<R>
+where
+    R: pallet_balances::Config + pallet_treasury::Config + frame_system::Config,
+    pallet_treasury::NegativeImbalanceOf<R>: From<NegativeImbalance<R>>,
+{
+    // this seems to be called for substrate-based transactions
+    fn on_unbalanceds(
+        mut fees_then_tips: impl Iterator<Item = Credit<R::AccountId, pallet_balances::Pallet<R>>>,
+    ) {
+        if let Some(fees) = fees_then_tips.next() {
+            // 100% of fees & tips goes to the treasury.
+            ResolveTo::<pallet_treasury::TreasuryAccountId<R>, pallet_balances::Pallet<R>>::on_unbalanced(fees);
+
+            if let Some(tip) = fees_then_tips.next() {
+                ResolveTo::<pallet_treasury::TreasuryAccountId<R>, pallet_balances::Pallet<R>>::on_unbalanced(tip);
+            }
+        }
+    }
+
+    // this is called from pallet_evm for Ethereum-based transactions
```

Comment:
remove the comment, we dont have pallet-evm here haha

Reply:
According to coverage, we don't use this function in any runtime. But let's keep it anyway (without the comment)